src/QAT/qat/purr/compiler/execution.py

```diff
@@ -68,7 +73,7 @@ def run_calibrations(self, qubits_to_calibrate=None):
         pass
 
     @abc.abstractmethod
-    def execute(self, instructions: List[Instruction]) -> Dict[str, Any]:
+    def execute_with_metrics(self, instructions: List[Instruction]) -> (Dict[str, Any], CompilationMetrics):
```

Comment:
"CompilationMetrics" sounds confusing when it's returned from a function that "execute"s instructions.
suggestion to rename `CompilationMetrics` to `QatMetric` as it's clearly not only about compilation metrics anymore now that you're adding new metrics.